text/0000-mixed-shadow-mode.md

```diff
@@ -11,80 +11,88 @@ pr: https://github.com/salesforce/lwc-rfcs/pull/49
-`@lwc/synthetic-shadow` package. Synthetic shadow DOM provides a consistent experience for LWC web
-components across the Salesforce-supported browser matrix.
+collectively referred to as synthetic Shadow DOM and are published to npm under the
+`@lwc/synthetic-shadow` package. Synthetic Shadow DOM provides a consistent experience for LWC web
+components across the Salesforce-supported browser matrix by implementing Shadow DOM semantics in
+older browsers that do not support it.
 
-Native Shadow DOM support has greatly increased since the start of the LWC project but applications
+Native Shadow DOM support has become more common since the start of the LWC project but applications
 continue to rely on the synthetic shadow polyfills to maintain backwards-compatibility. The removal
-of these polyfills can result in broken experiences due to dependencies on certain "escape hatches"
-that are not possible in a native Shadow DOM environment (e.g., instrumentation, styling, etc, which
-are discussed below).
+of these polyfills can result in broken experiences for existing applications due to dependencies on
+certain "escape hatches" that are not possible in a native Shadow DOM environment (e.g.,
+instrumentation, styling, etc).
 
 Applications that wish to use LWC components in a native shadow context simply need to omit the
-inclusion of the `@lwc/synthetic-shadow` polyfill; however, this is not a realistic option for
-large applications that cannot afford a rewrite.
+inclusion of the `@lwc/synthetic-shadow` polyfill; however, this all-or-nothing approach is not a
+realistic option for large applications that cannot afford a rewrite.
 
 This proposal opens up an incremental migration path for applications moving to native Shadow DOM by
-allowing the usage of both native and synthetic Shadow DOM in the same application.
+enabling the usage of both native and synthetic Shadow DOM in the same application.
 
 ## Detailed design
 
 Up until now, the two choices of shadow semantics have been synthetic mode and native mode. All the
 components in an application run in either synthetic mode or native mode, depending on whether the
 synthetic shadow polyfill (i.e., `@lwc/synthetic-shadow`) is applied. Mixed mode introduces the
-ability for a component to access native Shadow DOM APIs, even if the synthetic shadow polyfill is
-applied. Mixed mode should be transparent to components and the framework should guarantee this
+ability for a component to access native Shadow DOM APIs, even while the synthetic shadow polyfill
+is applied. Mixed mode should be transparent to components and the framework should guarantee this
 through integration testing, which is further discussed below.
 
 ### Opting in
 
-A component can signal that it prefers native Shadow DOM by setting a static property called
-`preferNativeShadow` to `true`.
+A component can signal that it supports both synthetic and native Shadow DOM by setting a static
+property called `shadowMode` to `any`.

```

Comment:
Bikeshed: The name `shadowMode` can easily be confused with the shadow root mode (`open` or `closed`).

Reply:
Maybe in the future we can call this "encapsulation mode" to avoid confusion? That's [what MDN calls it](https://developer.mozilla.org/en-US/docs/Web/API/Element/attachShadow#parameters).

Reply:
Yeah, [I also had this concern](https://github.com/salesforce/lwc-rfcs/pull/49#discussion_r645177988) but I couldn't come up with a better name. I used `shadowSupportMode` internally, what does everyone think about that?

Reply:
What about reusing `renderMode` from light dom?

Reply:
I think it would be overloading `renderMode` a bit too much. Also ideally in the far future we will deprecate `shadowSupportMode` (or whatever it's called), but we will still support `renderMode`.

Reply:
Changed to `shadowSupportMode`